thorlcr/activities/nsplitter/thnsplitterslave.cpp

```diff
@@ -209,6 +209,8 @@ class NSplitterSlaveActivity : public CSlaveActivity, implements ISharedSmartBuf
     {
         // NB: called by 1st output to start()
         CriticalBlock block(prepareInputLock);
+        if (writeAheadException)
+            throw LINK(writeAheadException);
```

Comment:
Should this only throw the exception if the error was generated in the prepare.  Should this have an additional check (0 == recsReady)?